x-pack/plugins/cloud_security_posture/public/pages/configurations/findings_flyout/findings_flyout.tsx

```diff
@@ -200,13 +202,16 @@ const FindingsTab = ({ tab, finding }: { finding: CspFinding; tab: FindingsTab }
   }
 };
 
-const isNativeCspFinding = (finding: CspFinding) =>
-  finding.data_stream.dataset === CSP_MISCONFIGURATIONS_DATASET;
-
-const MissingFieldsCallout = ({ finding }: { finding: CspFinding }) => {
+export const MissingFieldsCallout = ({
+  finding,
+}: {
+  finding: CspFinding | CspVulnerabilityFinding;
+}) => {
   const { euiTheme } = useEuiTheme();
   const datasetDisplayName =
-    getDatasetDisplayName(finding.data_stream.dataset) || finding.data_stream.dataset;
+    finding.observer?.vendor ||
```

Comment:
nit: what about encapsulating all this logic in getDatasetDisplayName ? because in the end all this logic is about getting display name

Reply:
plus you'll remove a bit of a copy paste from [overview_tab.tsx](https://github.com/elastic/kibana/pull/194362/files#diff-bc0f4846034a26fb866e47568562daad437e905399085d554e900225b63d8d21)